tools/omnissm/pkg/servicectl/service_windows.go

```diff
@@ -0,0 +1,121 @@
+// +build windows
+
+package servicectl
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/mgr"
+
+	"github.com/pkg/errors"
+)
+
+func openService(name string) (*mgr.Service, error) {
+	m, err := mgr.Connect()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to open windows service manager")
+	}
+	svc, err := m.OpenService(name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to open windows service %s", name)
+	}
+	return svc, nil
+}
+
+// NOTE: on windows this will only control existing services
+func newService(name string) (Service, error) {
+	svc, err := openService(name)
+	if err != nil {
+		return nil, err
+	}
+	// close service handle until we need it later
+	svc.Close()
+	return &windows{name}, nil
+}
+
+type windows struct {
+	name string
+}
+
+func (w *windows) wait(s *mgr.Service, timeout time.Duration, desired svc.State) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	for {
+		st, err := s.Query()
+		if err != nil {
+			return errors.Wrapf(err, "unable to query status of service %s", w.name)
+		}
+		if st.State == desired {
+			return nil
+		}
+		//return errors.Errorf("service %s is in state %v, not %v as requested", w.name, st.State, desired)
+		fmt.Printf("looping %v ms\n", st.WaitHint)
```

Comment:
do you want to print the hint or the actual value? for an observer seeing 0 and then get a 10s delay assumes knowledge of the code, ie. is it operational or debugging. if the later do we want to just omit?

Reply:
Thanks, this was debug that I had missed. Removed now.